### PR TITLE
GH Action for build and deploy website

### LINKED ISSUE #316: Switch to GH Actions for build and deploy?
Currently the MEI website is build through `gh-pages` which is enabled by the `gh-pages` gem in our Gemfile. However, this has the disadvantage that we can only use gems that are whitelisted and are limited to the fixed versions supported by GitHub.

Another possibility for building jekyll sites is through `GH Actions`, as described on the [official jekyll site](https://jekyllrb.com/docs/continuous-integration/github-actions/). Switching to `GH Actions` would allow to rely on the newest version of jekyll available as well as to include basically any gem we might want. This workflow is also quite common, as there are e.g. various GH Actions already available on [GitHub marketplace](https://github.com/marketplace?type=actions&query=jekyll+). This might also solve the discussion/problem around the `Gemfile.lock` in the context of building the site locally, see #93.

I'm not entirely sure if this approach would be feasible for MEI and if reusing an existing GH Action would be the best approach here. Maybe our GH Actions master @musicEnfanthen does have some advice?

### PR DESCRIPTION
**Pull request by doerners**
Set up a GH Action for building and deploying the website without gh-pages gem. Further updated jekyll in gemfile.

Co-authored-by: @rettinghaus 
Co-authored-by: @riedde 

Closes #316

### CODE REVIEW

**Comment by musicEnfanthen** [comment]
Thank you @doerners and co-authors for trying this workflow from https://jekyllrb.com/docs/continuous-integration/github-actions/

Before we go into details of the implementation, seems like the CSS has gone or is not recognized in the build output? See https://doerners.github.io/music-encoding.github.io/

**Comment by doerners** [comment]
That is true! However, when building the website locally styling get's rendered correctly. For the build on the fork the rendering is there if in the `config.yml` the `style_baseurl` is filled. Since some variables refer to that url somehow the site urls aren't build correctly. We couldn't figure out if it just applied to the build process for the fork.

**Comment by musicEnfanthen** [comment]
As mentioned in #93, there are reasons for sticking with the Gemfile.lock: https://bundler.io/guides/rationale.html

File: .github/workflows/jekyll.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Testing the GitHub Pages publication
+
+on:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (doerners) [reply .github/workflows/jekyll.yml]: Done :)

File: Gemfile.lock [hunk 2]
```diff
@@ -1,292 +1,73 @@
...
-      unf_ext
-    unf_ext (0.0.8)
     unicode-display_width (1.8.0)
     webrick (1.7.0)
-    zeitwerk (2.5.4)
 
 PLATFORMS
   arm64-darwin-21
-  x86_64-darwin-19
-  x86_64-linux
```
[the remark this answers is not in the document]
Comment (doerners) [reply Gemfile.lock]: done :)

**Comment by bwbohl** [comment]
> As mentioned in #93, there are reasons for sticking with the Gemfile.lock: https://bundler.io/guides/rationale.html

Only reason I can imagine is that GitHub dependabot uses them `Gemfile.lock` to identify dependencies and check them for security issues.

**Comment by ahankinson** [comment]
@bwbohl Gemfile.lock is used to determine the exact version of the packages installed. It is needed to exactly reproduce a known working configuration since the Gemfile only gives ranges,

**Review by musicEnfanthen — APPROVED** [review]

**Comment by musicEnfanthen** [comment]
Thanks for adding the lock file back, @doerners

**Comment by musicEnfanthen** [comment]
OK, Houston, ready to go :)

**Comment by rettinghaus** [comment]
@musicEnfanthen Thanks for merging, but the page is not only build with the new action but also still with the old [GitHub pages bot](https://github.com/music-encoding/music-encoding.github.io/actions/workflows/pages/pages-build-deployment). We still have to switch the website to `gh-pages` branch.

**Comment by musicEnfanthen** [comment]
Yes, I already did, thanks @rettinghaus

**Comment by musicEnfanthen** [comment]
The latest build run takes the gh-pages branch as source: https://github.com/music-encoding/music-encoding.github.io/actions/runs/3221577452

**Comment by bwbohl** [comment]
> @bwbohl Gemfile.lock is used to determine the exact version of the packages installed. It is needed to exactly reproduce a known working configuration since the Gemfile only gives ranges,
> 
> it also locks all the versions of dependencies too, while the Gemfile only lists the top level gems. It uses a hash too so that you can be fairly certain it would break if someone malicious started mucking about with version tags.

@ahankinson thanks for clarifying, and bringing to my attention that my comment was a bit short and unclear about what I was intending to say, being: I think it would be wise to stick to a Gemfile.lock for exactly those reasons. :-p

**Comment by musicEnfanthen** [comment]
Just for clarity: This new process only replaces the **build process** of the website, meaning Jekyll site is now build via the customized GH action, giving more options to include Jekyll plugins (like e.g., Jekyll Scholar) that are blacklisted when building jekyll sites with Github pages. Nonetheless, we still rely on Github pages for the deployment process, since GH pages delivers the website now (without building it) from the gh-pages branch where it is pre-built with our action.

**Comment by bwbohl** [comment]
The build result is being pushed to the branch `gh-pages` which is the branch that is being deployed, apparently the buildproecss generates the [.nojekyll](https://github.com/music-encoding/music-encoding.github.io/blob/gh-pages/.nojekyll) file that you, @rettinghaus, were missing ;-)

**Pull request merged**